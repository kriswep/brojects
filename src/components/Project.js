import React from 'react';
import { HeaderTabs, Tab } from 'react-mdl';

const Project = ({ projects = [], currentProject, onChangeProject }) => (
  <HeaderTabs ripple activeTab={currentProject} onChange={onChangeProject}>
    {/*onChange={(tabId) => this.setState({ activeTab: tabId })}*/}
    {projects.map((project) => {
      return (
        <Tab id={project.id} key={project.id}>{project.name}</Tab>
      );
    })}
  </HeaderTabs>
)

export default Project;

            {/*<HeaderTabs ripple  >
                            <Tab>Tab1</Tab>
                            <Tab>Tab2</Tab>
                            <Tab>Tab3</Tab>
                            <Tab>Tab4</Tab>
                            <Tab>Tab5</Tab>
                            <Tab>Tab6</Tab>
                        </HeaderTabs>*/}