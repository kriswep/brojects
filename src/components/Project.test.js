import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Project from './Project';

describe('Project', () => {

  it('renders without crashing', () => {
    const wrapper = shallow(<Project />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders as expected', () => {
    const projects = [{
      id: 'projectID1',
      name: 'project/1'
    },{
      id: 'repoID2',
      name: 'project/2'
    },
    ]
    
    const wrapper = shallow(<Project projects={projects} currentProject={0}/>);

    expect(toJson(wrapper)).toMatchSnapshot();    
  });
});