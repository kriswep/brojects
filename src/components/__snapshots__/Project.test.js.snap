exports[`Project renders as expected 1`] = `
<HeaderTabs
  activeTab={0}
  ripple={true}>
  <Tab
    id="projectID1"
    style={Object {}}>
    project/1
  </Tab>
  <Tab
    id="repoID2"
    style={Object {}}>
    project/2
  </Tab>
</HeaderTabs>
`;

exports[`Project renders without crashing 1`] = `
<HeaderTabs
  ripple={true} />
`;
